### PR TITLE
CMR-9410: Missing MetadataSpecification handling

### PR DESCRIPTION
**Pull request by sxu123**

### CODE REVIEW

File: ingest-app/src/cmr/ingest/api/generic_documents.clj [hunk 1]
```diff
@@ -129,10 +130,15 @@
...
-             (not= (common-concepts/get-concept-type-of-draft concept-type) spec-key))
-      (throw (UnsupportedOperationException. (format "%s version %s is not supported." spec-key spec-version)))
+    (if (or (not spec-key)
+            (and (not= concept-type spec-key)
+                 (not= (concept-type draft-concept->spec-map) spec-key)
+                 (not= (common-concepts/get-concept-type-of-draft concept-type) spec-key)))
+      (errors/throw-service-error
+       :invalid-data
+       (if-not spec-key
+         "MetadataSpecification is missing"
```
Comment (eereiter) [inline ingest-app/src/cmr/ingest/api/generic_documents.clj:140]: Can you change the string to 

"The MetadataSpecification schema element is missing from the record being ingested."

Comment (sxu123) [reply ingest-app/src/cmr/ingest/api/generic_documents.clj]: Done

File: ingest-app/src/cmr/ingest/api/generic_documents.clj [hunk 2]
```diff
@@ -129,10 +130,15 @@
...
-      (throw (UnsupportedOperationException. (format "%s version %s is not supported." spec-key spec-version)))
+    (if (or (not spec-key)
+            (and (not= concept-type spec-key)
+                 (not= (concept-type draft-concept->spec-map) spec-key)
+                 (not= (common-concepts/get-concept-type-of-draft concept-type) spec-key)))
+      (errors/throw-service-error
+       :invalid-data
+       (if-not spec-key
+         "The MetadataSpecification schema element is missing from the record being ingested."
+         (str spec-key " version " spec-version " are not supported")))
```
Comment (eudoroolivares2016) [inline ingest-app/src/cmr/ingest/api/generic_documents.clj:141]: Can we add a test for if the version is incorrect

Comment (sxu123) [reply ingest-app/src/cmr/ingest/api/generic_documents.clj:141]: If you look at the code I replaced, I was only adding the case when spec-key doesn't exist,  which is when MetadataSpecification doesn't exist. the other part of the functionality is carried over.  That is also only checking the spec key.  version is not being checked. It's just returned in the message, together with the invalid spec-key.

Comment (sxu123) [reply ingest-app/src/cmr/ingest/api/generic_documents.clj:141]: added one more test when spec-key is wrong.

**Review by eudoroolivares2016 — APPROVED** [review]

File: ingest-app/src/cmr/ingest/api/generic_documents.clj (hunk 2 above)
Comment (eudoroolivares2016) [reply ingest-app/src/cmr/ingest/api/generic_documents.clj:141]: Phrased that wrong you are correct. We do handle the version already, and with this we can now handle the wrong "Name" thanks for adding test.

**Review by eereiter — APPROVED** [review]

**Pull request merged**